node/resolvers/Mutations/Users.ts

```diff
@@ -612,3 +642,45 @@ const Users = {
 }
 
 export default Users
+
+async function impersonateUser(
+  storefrontPermissionsClient: any,
+  id: string,
+  logger: any,
+  accountName: string,
+  storefrontPermissions: {
+    costcenter: { value: string }
+    userId: { value: string }
+    organization: { value: string }
+    storeUserEmail: { value: string }
+  }
+) {
+  const result = await storefrontPermissionsClient
+    .impersonateUser({ id })
+    .catch((error: any) => {
+      logger.error({
+        error,
+        message: 'impersonateUser-impersonateUserError',
+      })
+
+      return { status: 'error', message: error }
```

Comment:
I believe we don't want to send the metrics if the call to impersonate fails, right? Or, if we do want to send metrics in error cases, we need to fix the `sendImpersonateB2BUserMetric` call further above to stop ignoring errors too.

Reply:
Right again.
Fix to not send metrics to analytics on impersonate errors